tests/baselines/reference/fixSignatureCaching.types

```diff
@@ -866,9 +866,9 @@ define(function () {
 >'UnknownMobile' : "UnknownMobile"
 
     isArray = ('isArray' in Array) ?
->isArray = ('isArray' in Array) ?        Array.isArray : function (value) { return Object.prototype.toString.call(value) === '[object Array]'; } : (arg: any) => arg is any[]
+>isArray = ('isArray' in Array) ?        Array.isArray : function (value) { return Object.prototype.toString.call(value) === '[object Array]'; } : (value: any) => boolean

```

Comment:
I think this change is correct because only the left hand side of the conditional is a type predicate. @ahejlsberg could you verify, then merge

Reply:
Something seems fishy here. If you reverse the order of the operands you get the old result. So apparently both types are subtypes of each other, or otherwise subtype reduction would consistently pick one of them.  I'm not sure why this is the case, since `(x: any) => x is any[]` should be a subtype of `(x: any) => boolean`. You may want to investigate.

Reply:
Added a test -- the return type seems to be `boolean` the other way around as well.

Reply:
Ok, looks good.